assignments/prototypes.js

```diff
@@ -7,21 +7,34 @@
   
   Each constructor function has unique properties and methods that are defined in their block comments below:
 */
-  
+
 /*
   === GameObject ===
   * createdAt
   * name
   * dimensions (These represent the character's size in the video game)
   * destroy() // prototype method that returns: `${this.name} was removed from the game.`
 */
-
+function GameObject(attributes) {
+  this.createdAt = attributes.createdAt;
+  this.name = attributes.name;
+  this.dimensions = attributes.dimensions;
+}
+GameObject.prototype.destroy = () => {
+  returns: `${this.name} was removed from the game.`;
+};
 /*
   === CharacterStats ===
   * healthPoints
   * takeDamage() // prototype method -> returns the string '<object name> took damage.'
   * should inherit destroy() from GameObject's prototype
 */
+function CharacterStats(attributes) {
+  this.healthPoints = attributes.healthPoints;
```

Comment:
CharacterStats needs to bind to the attributes of its parent constructor GameObject. So you need something like 
```
GameObject.call(this, attributes);
```
And before the `CharacterStats.prototype.takeDamage`, you need something like:
```
CharacterStats.prototype = Object.create(GameObject.prototype);
```

This is why your mage and archer etc are not working. 

Here is the codepen with examples of constructor function inheritance:
https://codepen.io/letanque/pen/YbYeMb